pkg/state/storage.go

```diff
@@ -136,18 +136,23 @@ func (st *Storage) normalizePath(path string) string {
 	if u != nil && (u.Scheme != "" || filepath.IsAbs(path)) {
 		return path
 	} else {
-		return st.JoinBase(path, runtime.GOOS)
+		return st.JoinBase(path)
 	}
 }
 
 // JoinBase returns an absolute path in the form basePath/relative
 // Helm's setFiles command does not support unescaped filepath separators (\) on Windows.
 // Instead, it requires double backslashes (\\) as filepath separators.
 // See https://github.com/helm/helm/issues/9537
-func (st *Storage) JoinBase(relPath, GOOS string) string {
+func (st *Storage) JoinBase(relPath string) string {
 	path := filepath.Join(st.basePath, relPath)
-	if GOOS == "windows" {
-		return strings.ReplaceAll(path, "\\", "\\\\")
-	}
 	return path
 }
+
+func (st *Storage) normalizeSetFilePath(path, goos string) string {
```

Comment:
why do need goos args? we can get it from `runtime.GOOS`?

Reply:
To make it easier to write unit tests